lib/logger.js

```diff
@@ -150,6 +167,52 @@ Logger.prototype._levelToMatcher = function _levelToMatcher(level) {
     }
 };
 
+/**
+ * Parses the provided logging level for a log component,
+ * matches it with the configured specially logged components.
+ *
+ * If the message should be logged, returns an object with bunyan log level
+ * and a specialized logger for the given component, otherwise returns undefined
+ *
+ * @param {String} level a logging level
+ * @returns {Object|undefined} corresponding bunyan log level and a specialized logger
+ * @private
+ */
+Logger.prototype._getComponentLogConfig = function(level) {
+    if (level && Object.keys(this._log_components).length) {
```

Comment:
It would be more efficient / simpler to just set up a probability of logging a specific levelPath, and look that up in an object. This is going to be called for *a lot* of trace / debug etc events that would otherwise not be logged, so making the decision on whether to probabilistically log fast and robust is important.